flytekit/core/data_persistence.py

```diff
@@ -254,6 +254,8 @@ def get(self, from_path: str, to_path: str, recursive: bool = False, **kwargs):
             return to_path
         except OSError as oe:
             logger.debug(f"Error in getting {from_path} to {to_path} rec {recursive} {oe}")
+            if not file_system.exists(from_path):
```

Comment:
is it possible to raise an error without an additional network call? e.g. are we able to check the error code (404) from `oe`?

Reply:
if not, that's fine

Reply:
Neat, but I've tried if `oe.errno == FileNotFoundError.errno`: and it failed to catch the `404` error.

I think it's ignorable wrt the performance issue, since it's already trapped in the exception handling process.